meshtastic/interfaces/ble/lifecycle_shutdown_runtime.py

```
@@ -0,0 +1,459 @@
+        safe_cleanup: Callable[[Callable[[], object], str], None] | None = None,
+    ) -> None:
+        """Close discovery resources and clear receive-loop intent."""
+        iface = self._iface
+        run_safe_cleanup = safe_cleanup or self._error_access.safe_cleanup
+        discovery_manager = iface._discovery_manager
+        iface._discovery_manager = None
+        if discovery_manager is not None:
+            run_safe_cleanup(discovery_manager.close, "discovery manager close")
+        iface._set_receive_wanted(want_receive=False)
+
+    def _shutdown_receive_thread(
```

Ruff (0.15.6)

[warning] Too many branches (20 > 12)

(PLR0912)